internal/controller/management_controller.go

```diff
@@ -1133,6 +1193,23 @@ func (r *ManagementReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		setupLog.Info("Validations are disabled, watcher for ProviderTemplate objects is set")
 	}
 
+	managedController.Watches(&kcm.PluggableProvider{}, handler.EnqueueRequestsFromMapFunc(func(context.Context, client.Object) []ctrl.Request {
```

Comment:
nit: might be added directly during construction of the `managedController` variable to improve readability